src/transformStateWithClones.js

```diff
@@ -8,6 +8,27 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const finalObject = [];
+  let modObject = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(modObject, action.extraData);
+      finalObject.push({ ...modObject });
```

Comment:
The push to `finalObject` is repeated in every case. To adhere to the DRY (Don't Repeat Yourself) principle, you should perform this action once after the switch statement.